File: Backend Codes/MATLAB final version/calcTieredRate.m
function [Cbuy]= calcTieredRate(prices, tierMax, load, daysInMonth)
    Cbuy = zeros(1,8760);
    hCount = 1;
    for m=1:12
        monthlyLoad = 0;
        for h=1:(24 * daysInMonth(m))
            monthlyLoad = monthlyLoad +load(hCount);
            if monthlyLoad < tierMax(1)
   
                Cbuy(hCount) = prices(1);
            elseif monthlyLoad < tierMax(2)
                 
                Cbuy(hCount) = prices(2);
            else
                Cbuy(hCount) = prices(3);
            end
            hCount = hCount+1;
        end
    end
end

File: src/samapy/samapy/pricing/calcTieredRate.py
import numpy as np


def calcTieredRate(tieredPrices, tierMax, Eload, daysInMonth):
    Cbuy = np.zeros(8760)
    hCount = 0

    for m in range(12):
        monthlyLoad = 0
        for h in range(24 * daysInMonth[m]):
            monthlyLoad += Eload[hCount]

            if monthlyLoad < tierMax[0]:
                Cbuy[hCount] = tieredPrices[0]
            elif monthlyLoad < tierMax[1]:
                Cbuy[hCount] = tieredPrices[1]
            else:
                Cbuy[hCount] = tieredPrices[2]

            hCount += 1

    return Cbuy